pkg/condition/conditions.go

```diff
@@ -37,10 +45,34 @@ type Conditions interface {
 	IsPodSGPEnabled() bool
 }
 
-func NewControllerConditions(dataStoreSyncFlag *bool, log logr.Logger) Conditions {
+var (
+	conditionWindowsIPAMEnabled = prometheus.NewSummaryVec(
+		prometheus.SummaryOpts{
```

Comment:
Why do we need a Summary metrics here? This represents a binary output 0/1 which should not require summary observations. 
Have you considered using https://prometheus.io/docs/concepts/metric_types/#gauge

Reply:
Changed it to use gauge.